lib/pat.c

```diff
@@ -5124,8 +5124,13 @@ grn_pat_check(grn_ctx *ctx, grn_pat *pat)
   GRN_OUTPUT_INT64(h->n_entries);
   GRN_OUTPUT_CSTR("curr_rec");
   GRN_OUTPUT_INT64(h->curr_rec);
-  GRN_OUTPUT_CSTR("curr_key");
-  GRN_OUTPUT_INT64(h->curr_key);
+  if (!pat_is_key_large(pat)) {
+    GRN_OUTPUT_CSTR("curr_key");
+    GRN_OUTPUT_INT64(h->curr_key);
+  } else {
+    GRN_OUTPUT_CSTR("curr_key_large");
+    GRN_OUTPUT_UINT64(h->curr_key_large);
+  }
```

Comment:
How about reversing the condition as follows? It will be easy to understand it.
```suggestion
  if (pat_is_key_large(pat)) {
    GRN_OUTPUT_CSTR("curr_key_large");
    GRN_OUTPUT_UINT64(h->curr_key_large)
  } else {;
    GRN_OUTPUT_CSTR("curr_key");
    GRN_OUTPUT_INT64(h->curr_key);
  }
```